docs/reference/configuration/configuration.md

```diff
@@ -1,48 +1,276 @@
 
-The default stack size of the main application thread is 4 kilobytes. This memory is dynamically allocated from the global heap. You can configure the stack size of the main thread by using the [Mbed config system](/docs/development/reference/configuration.html). To reduce the stack size from 4K to 2K, create an `mbed_app.json` file for your project with the following content:
+<span class="notes">**Note:** Throughout this document, "library" means any reusable piece of code within its own directory.</span>
+
+<span class="notes">**Note:** In prior releases, the configuration system provided a method for adding custom targets. The Mbed OS tools now look for custom targets in a file named `custom_targets.json` in the root of an application and treat custom targets the same as [Mbed targets](/docs/development/tools/adding-and-configuring-targets.html).</span>
+
+### Examining available configuration parameters
+
+Mbed CLI includes a command for listing and explaining the compile time configuration, `mbed compile --config`. This command prints a summary of configuration parameters, such as:
+
+```
+Configuration parameters
+------------------------
+cellular.random_max_start_delay = 0 (macro name: "MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY")
+cellular.use-apn-lookup = 1 (macro name: "MBED_CONF_CELLULAR_USE_APN_LOOKUP")
+configuration-store.storage_disable = 0 (macro name: "CFSTORE_STORAGE_DISABLE")
+drivers.uart-serial-rxbuf-size = 256 (macro name: "MBED_CONF_DRIVERS_UART_SERIAL_RXBUF_SIZE")
+drivers.uart-serial-txbuf-size = 256 (macro name: "MBED_CONF_DRIVERS_UART_SERIAL_TXBUF_SIZE")
+events.present = 1 (macro name: "MBED_CONF_EVENTS_PRESENT")
+events.shared-dispatch-from-application = 0 (macro name: "MBED_CONF_EVENTS_SHARED_DISPATCH_FROM_APPLICATION")
+events.shared-eventsize = 256 (macro name: "MBED_CONF_EVENTS_SHARED_EVENTSIZE")
+events.shared-highprio-eventsize = 256 (macro name: "MBED_CONF_EVENTS_SHARED_HIGHPRIO_EVENTSIZE")
+events.shared-highprio-stacksize = 1024 (macro name: "MBED_CONF_EVENTS_SHARED_HIGHPRIO_STACKSIZE")
+events.shared-stacksize = 1024 (macro name: "MBED_CONF_EVENTS_SHARED_STACKSIZE")
+<output truncated for brevity>
+```
+
+You may find that configuration naming does not always describe the parameter's purpose adequately. Use the `-v` switch to include the help text defined with the configuration parameter, where the value of the configuration parameter is defined, and other details. The command `mbed compile --config -v` in the same application as above prints:
```

Comment:
Query: Do we create the configuration naming? If so, it should always describe the parameter's purpose adequately. Is it OK to delete the first sentence and start with "Use the..."?

Reply:
> Do we create the configuration naming? If so, it should always describe the parameter's purpose adequately. 

Not always, and ideally yes, but the behavior can sometimes be hard to fit in a short string, so the help text is used to explain more thoroughly.

> Is it OK to delete the first sentence and start with "Use the..."?

Honestly, yeah that's a much better solution 😄 Let's do that!